From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Thu, 13 Aug 2026 16:48:09 +0200
Subject: ARM: dts: sun5i: pocketbook-touch-lux-3: Smooth CIE compensated
 backlight

Drop the 11-step percentage brightness-levels table; without it the
pwm-backlight driver generates a fine-grained CIE 1931 perceptually
uniform brightness table sized to the PWM resolution, same as on the
PinePhone. lth-brightness can be added/tuned (also via sysfs) if the
lowest steps turn out to be below the LED turn-on threshold.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm/boot/dts/allwinner/sun5i-a13-pocketbook-touch-lux-3.dts | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/allwinner/sun5i-a13-pocketbook-touch-lux-3.dts b/arch/arm/boot/dts/allwinner/sun5i-a13-pocketbook-touch-lux-3.dts
index 111111111111..222222222222 100644
--- a/arch/arm/boot/dts/allwinner/sun5i-a13-pocketbook-touch-lux-3.dts
+++ b/arch/arm/boot/dts/allwinner/sun5i-a13-pocketbook-touch-lux-3.dts
@@ -26,8 +26,10 @@ backlight {
 		compatible = "pwm-backlight";
 		pwms = <&pwm 0 50000 PWM_POLARITY_INVERTED>;
 		enable-gpios = <&pio 1 4 GPIO_ACTIVE_HIGH>; /* PB4 */
-		brightness-levels = <0 10 20 30 40 50 60 70 80 90 100>;
-		default-brightness-level = <8>;
+		/* no brightness-levels: the driver generates a fine
+		 * CIE 1931 perceptually uniform table (as on the
+		 * PinePhone); tune lth-brightness if the first steps
+		 * turn out dark */
 		power-supply = <&reg_vcc3v3>;
 	};
 
-- 
Armbian

